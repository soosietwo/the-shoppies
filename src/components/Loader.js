import React from "react";
import {
  ResourceItem,
  TextContainer,
  SkeletonDisplayText,
  SkeletonBodyText,
  ResourceList,
  SkeletonThumbnail,
  Stack,
} from "@shopify/polaris";

const Loader = ({ count }) => {
  return (
    <ResourceList
      items={[...Array(count).keys()]}
      renderItem={() => (
        <ResourceItem>
          <Stack>
            <SkeletonThumbnail size="large" />
            <Stack.Item fill>
              <TextContainer>
                <SkeletonDisplayText size="small" />
                <SkeletonBodyText />
              </TextContainer>
            </Stack.Item>
          </Stack>
        </ResourceItem>
      )}
    />
  );
};

export default Loader;
