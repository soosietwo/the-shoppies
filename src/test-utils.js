// test-utils.js
import React from "react";
import { render } from "@testing-library/react";
import { AppProvider, Frame } from "@shopify/polaris";
import en from "@shopify/polaris/locales/en.json";
import Store from "./store";

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const AllTheProviders = ({ children }) => {
  return (
    <AppProvider i18n={en}>
      <Store>
        <Frame>{children}</Frame>
      </Store>
    </AppProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
