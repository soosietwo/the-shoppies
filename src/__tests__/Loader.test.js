import React from "react";
import { render } from "../test-utils";
import Loader from "../components/Loader";

describe("Loader", () => {
  it("renders and matches snapshot", () => {
    const { container } = render(<Loader />);

    expect(
      container.querySelector(".Polaris-ResourceList__ResourceListWrapper")
    ).toMatchSnapshot();
  });

  it("renders the correct number of cards", () => {
    const { container } = render(<Loader count={5} />);

    expect(
      container.querySelectorAll(".Polaris-ResourceItem__ListItem")
    ).toHaveLength(5);
  });
});
