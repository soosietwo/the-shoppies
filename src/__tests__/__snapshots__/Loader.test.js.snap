// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders and matches snapshot 1`] = `
<div
  class="Polaris-ResourceList__ResourceListWrapper"
>
  <ul
    aria-live="polite"
    class="Polaris-ResourceList"
  >
    <li
      class="Polaris-ResourceItem__ListItem"
    >
      <div
        class="Polaris-ResourceItem__ItemWrapper"
      >
        <div
          class="Polaris-ResourceItem"
        >
          <button
            aria-label="View details for item"
            class="Polaris-ResourceItem__Button"
            tabindex="0"
          />
          <div
            class="Polaris-ResourceItem__Container"
          >
            <div
              class="Polaris-ResourceItem__Content"
            >
              <div
                class="Polaris-Stack"
              >
                <div
                  class="Polaris-Stack__Item"
                >
                  <div
                    class="Polaris-SkeletonThumbnail Polaris-SkeletonThumbnail--sizeLarge"
                  />
                </div>
                <div
                  class="Polaris-Stack__Item Polaris-Stack__Item--fill"
                >
                  <div
                    class="Polaris-TextContainer"
                  >
                    <div
                      class="Polaris-SkeletonDisplayText__DisplayText Polaris-SkeletonDisplayText--sizeSmall"
                    />
                    <div
                      class="Polaris-SkeletonBodyText__SkeletonBodyTextContainer"
                    >
                      <div
                        class="Polaris-SkeletonBodyText"
                      />
                      <div
                        class="Polaris-SkeletonBodyText"
                      />
                      <div
                        class="Polaris-SkeletonBodyText"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
